### PR TITLE
Feature/transactional db config

### PR DESCRIPTION
**Pull request by akebi**
Changes include
1. Support added for auto-generation of migrations
2. Branching out of migrations to support multiple schemas
3. Added PostgreSQL db connection to options (in addition to SQLite)
4. PyTest fixtures for database creation tests
5. Integration with CircleCi 2.0

### CODE REVIEW

File: dbutils.py [hunk 1]
```diff
@@ -167,3 +177,4 @@ def conn_uri_factory(conf_file, db_adapter):
 
 # db inspector
```
Comment (NdagiStanley) [inline dbutils.py:178]: Missing something here?

File: models/transactional.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ class ParticipantProgram(Base):
     program_cip = Column(Integer, ForeignKey('program.program_cip'), primary_key=True)
     entry_date = Column(Date, nullable=False)
     exit_date = Column(Date)
-    enrolled = Column(Boolean, nullable=False, default=False)
+    enrolled = Column(Boolean, nullable=False, default=True)
```
Comment (NdagiStanley) [inline models/transactional.py:42]: Why the change to have this as `True`?

**Review by NdagiStanley — COMMENTED** [review]
I have noted that there are three sets of files that have been renamed but with little or no difference. Any reason as to why? Just curious.

**Comment by NdagiStanley** [comment]
Nitpicking here 😄 
Have a look at the file endings.

Consider this:
screenshot 2017-07-10 20 42 48
and this:
screenshot 2017-07-10 20 43 05

I find it best to just have one empty line as the file terminator.

**Comment by NdagiStanley** [comment]
Lastly please let's have the merge conflicts resolved.

**Comment by akebi** [comment]
@NdagiStanley - noted for convention/consistency, let's maintain single line file terminator

File: models/transactional.py (hunk 2 above)
Comment (akebi) [reply models/transactional.py:42]: From what I could see, participants are initially captured because they enroll & the status may change because of drop-outs from programs etc.

File: dbutils.py (hunk 1 above)
Comment (akebi) [reply dbutils.py:178]: db inspector is not yet implemented @NdagiStanley

Comment (NdagiStanley) [reply dbutils.py:178]: Cool

File: models/transactional.py (hunk 2 above)
Comment (NdagiStanley) [reply models/transactional.py:42]: Got it. 👍

File: dbutils.py (hunk 1 above)
Comment (NdagiStanley) [reply dbutils.py:178]: Nitpick: One line terminator

File: models/transactional.py (hunk 2 above)
Comment (NdagiStanley) [reply models/transactional.py:42]: Makes sense, LGTM.

**Comment by akebi** [comment]
@NdagiStanley, good to go - all merge conflicts resolved

**Comment by akebi** [comment]
all tests passing after alembic update to 0.9.3 as well. @NdagiStanley good to go with merge to master
